tests/specs/common.blocks/i-model/__field/i-model__field.spec.js

```diff
@@ -0,0 +1,359 @@
+
+        it('should return unprocessed value', function() {
+            BEM.MODEL.decl('model', { field: { type: '', preprocess: function(val) { return val + '!' } } });
+
+            var model = BEM.MODEL.create('model');
+
+            model.set('field', 'val');
+
+            expect(model.get('field', 'raw')).to.equal('val');
+        });
+
+        it('should pre-process value from isEqual method', function() {
+            BEM.MODEL.decl('model', { field: { type: '', preprocess: function(val) { return val + '!' } } });
+
+            var model = BEM.MODEL.create('model', { field: 'val' });
+
+            expect(model.isEqual({ field: 'val' })).to.equal(true);
+        });
+    });
+
+    describe('calculate', function() {
+        it('should return calculated value', function() {
+            BEM.MODEL.decl('model', {
+                field: {
+                    type: '',
+                    calculate: function() { return 'val' }
+                }
+            });
+
+            var model = BEM.MODEL.create('model');
+
+            expect(model.get('field')).to.equal('val');
+        });
+
+        it('should return calculated value by other fields', function() {
+            BEM.MODEL.decl('model', {
+                field1: '',
+                field2: {
+                    type: '',
+                    calculate: function() { return this.get('field1') + '!' }
+                }
+            });
+
+            var model = BEM.MODEL.create('model', { field1: 'val' });
+
+            expect(model.get('field2')).to.equal('val!');
+        });
+
+        it('should calculated value by one filed and return it', function() {
+            BEM.MODEL.decl('model', {
+                field1: '',
+                field2: {
+                    type: '',
+                    calculate: function(val) { return val + '!' },
+                    dependsFrom: ['field1']
+                }
+            });
+
+            var model = BEM.MODEL.create('model', { field1: 'val' });
+
+            expect(model.get('field2')).to.equal('val!');
+        });
+
+        it('should calculated value by fileds and return it', function() {
+            BEM.MODEL.decl('model', {
+                field1: '',
+                field2: '',
+                field3: {
+                    type: '',
+                    calculate: function(vals) { return vals.field1 + vals.field2 },
+                    dependsFrom: ['field1', 'field2']
+                }
+            });
+
+            var model = BEM.MODEL.create('model', { field1: 'val1', field2: 'val2' });
+
+            expect(model.get('field3')).to.equal('val1val2');
+        });
+    });
+
+    describe('validation', function() {
+        it('by default is valid', function() {
+            BEM.MODEL.decl('model', { field: '' });
+
+            var model = BEM.MODEL.create('model');
+
+            expect(model.isValid()).to.equal(true);
+        });
+
+        it('should define from function', function() {
```

Comment:
validation should define from function -> validation could be defined by function